### DISCUSSION TITLE
How to change an element inside a RankTwoTensor?

### DISCUSSION THREAD

**Discussion by pharshaad** [idaholab/moose#16981] [Q&A General] [answered]
Hi all,

I was wondering how I can change a particular element inside a RankTwoTensor. To give a little perspective,  I have a material that computes an ADRankTwoTensor consisting of the gradient of some variable. More specifically, 

```
void
ADComputeSomeQuantity::computeQpProperties()
{
  ADRankTwoTensor A((*_grad_ref[0])[_qp], (*_grad_ref[1])[_qp], (*_grad_ref[2])[_qp]);
}
```

The goal is to inverse this tensor to get something like `_some_quantity[_qp] = A.inverse();`

Since I'm working in 2-D at the moment, the 3rd row and the 3rd column of tensor A is zero which makes it singular. I was wondering if there's a way to access a particular element to change for example A_33 to something non-zero. 

Thanks,
Ari

**Comment by hugary1995** [comment, marked as answer]
You can do 
`A(2, 2) = 1;`
MOOSE RankTwoTensor doxygen can be found here: [https://mooseframework.inl.gov/docs/doxygen/moose/classRankTwoTensorTempl.html](https://mooseframework.inl.gov/docs/doxygen/moose/classRankTwoTensorTempl.html).
RankTwoTensor extends libMesh's TypeTensor:
[https://libmesh.github.io/doxygen/classlibMesh_1_1TypeTensor.html](https://libmesh.github.io/doxygen/classlibMesh_1_1TypeTensor.html).

In your case, what you are looking for is the `operator()` in TypeTensor.

**Reply by pharshaad** [reply]
Thanks for the tip Gary! Much appreciated.
